OpenRA.Mods.Common/Widgets/Logic/Lobby/ClientTooltipLogic.cs

```diff
@@ -79,10 +79,15 @@ public ClientTooltipLogic(Widget widget, TooltipContainerWidget tooltipContainer
 			var ping = orderManager.LobbyInfo.PingFromClient(client);
 			latency.GetText = () => LobbyUtils.LatencyDescription(ping);
 			latency.GetColor = () => LobbyUtils.LatencyColor(ping);
+
 			var address = orderManager.LobbyInfo.ClientWithIndex(clientIndex).IpAddress;
-			if (clientIndex == orderManager.LocalClient.Index && UPnP.NatDevice != null
-				&& address == IPAddress.Loopback.ToString())
-				address = UPnP.NatDevice.GetExternalIP().ToString();
+			if (clientIndex == orderManager.LocalClient.Index && address == IPAddress.Loopback.ToString())
+			{
+				var externalIP = IPAddress.None;
+				if (UPnP.NatDevice != null && IPAddress.TryParse(UPnP.NatDevice.GetExternalIP().ToString(), out externalIP))
```

Comment:
This doesn't make much sense to me. You are converting to string and then parsing the string back into an IP address only to save it as string below.